src/graph/graph.h

```diff
@@ -137,6 +137,15 @@ void Graph_CreateNode(
 	Node *n
 );
 
+// Sets a node in the graph - Used for deserialization of graph.
+// Return the set node.
+void Graph_SetNode(
```

Comment:
As a broad comment, I would prefer if functions used exclusively for serialization lived in a different file, preferably within the encode/decode folders. Their names should also indicate that they're serialization functions.